flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/runtime/stream/sql/FunctionITCase.java

```diff
@@ -1542,6 +1542,19 @@ void testUsingAddJar() throws Exception {
                 "drop function lowerUdf");
     }
 
+    @Test
+    void testUsingAddJarWithCheckpointing() throws Exception {
+        env().enableCheckpointing(100);
```

Comment:
NIT: I guess we could check that the class loader on the Current thread  is the same before and after the  env.executeSql. Using the  Thread.currentThread().getContextClassLoader();

Reply:
Assertions added for verifying the class loader before and after.